app/src/main/java/com/bilal/cactus_notes_project/ui/SignupActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.bilal.cactus_notes_project.ui
+
+import android.os.Bundle
+import android.util.Patterns
+import androidx.appcompat.app.AppCompatActivity
+import com.bilal.cactus_notes_project.R
+import com.bilal.cactus_notes_project.databinding.ActivitySignupBinding
+
+
+class SignupActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignupBinding
+    private val userNameRegex = "^[a-z0-9_]*$".toRegex()
+    private val passwordRegex = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=]).+$".toRegex()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignupBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        binding.btnSignup.setOnClickListener {
+            val email = binding.textInputEditTextEmail.text
+            val userName = binding.textInputEditTextUserName.text
+            val password = binding.textInputEditTextPassword.text
+            if (checkEmailAddress(email.toString()) and checkUserName(userName.toString()) and checkPassword(
+                    password.toString()
+                )
+            ) {
+                //TODO("Go to login activity")
```

Comment:
Better syntax for TODO comments is:

// TODO: Go to LoginActivity

Reply:
DONE